onnx/defs/schema.cc

```diff
@@ -362,14 +362,103 @@ OpSchema& OpSchema::Attr(const Attribute& attr) {
 }
 
 OpSchema& OpSchema::Attr(
-    const char* name,
-    const char* description,
+    const std::string& name,
+    const std::string& description,
     AttributeProto::AttributeType type,
     bool required) {
```

Comment:
It's a very confusing prototype and the way it looks in the declarations (`, false`) is very easy to confuse between required vs default value. Can we replace `bool required` with an enum OPTIONAL or REQUIRED? Then it'd be more obvious and readable

Reply:
:). I'm assuming this (,false) calls should be temporary thing as well as we walking thru them and assign default values accordingly. However, you're right. It's not that readable and a little bit confusing, so let me create a "bool Optional = false;" and refer to this. Sounds good?